launcher/ext/innoextract/CMakeLists.txt

```diff
@@ -0,0 +1,538 @@
+
+# Components
+option(USE_ARC4 "Build ARC4 decryption support" ON)
+
+# Optional dependencies
+option(USE_LZMA "Build LZMA decompression support" ON)
+option(USE_DYNAMIC_UTIMENSAT "Dynamically load utimensat if not available at compile time" OFF)
+
+# Alternative dependencies
+set(WITH_CONV CACHE STRING "The library to use for charset conversions")
+
+# Build types
+option(DEBUG_EXTRA "Expensive debug options" OFF)
+option(SET_WARNING_FLAGS "Adjust compiler warning flags" ON)
+option(SET_NOISY_WARNING_FLAGS "Enable noisy compiler warnings" OFF)
+option(SET_OPTIMIZATION_FLAGS "Adjust compiler optimization flags" ON)
+suboption(USE_LDGOLD "Use the Gold linker" BOOL ${SET_OPTIMIZATION_FLAGS})
+set(default_FASTLINK OFF)
+if(DEVELOPER OR CONTINUOUS_INTEGRATION)
+	set(default_FASTLINK ON)
+endif()
+suboption(FASTLINK "Optimize (incremental) linking speed" BOOL ${default_FASTLINK})
+set(default_USE_LTO OFF)
+if(SET_OPTIMIZATION_FLAGS AND NOT FASTLINK)
+	set(default_USE_LTO ON)
+endif()
+suboption(USE_LTO "Use link-time code generation" BOOL ${default_USE_LTO})
+suboption(WERROR "Turn warnings into errors" BOOL ${CONTINUOUS_INTEGRATION})
+suboption(CXX_STD_VERSION "Maximum C++ standard version to enable" STRING 2017)
+if(DEVELOPER OR CMAKE_BUILD_TYPE STREQUAL "Debug")
+	set(default_DEBUG ON)
+else()
+	set(default_DEBUG OFF)
+endif()
+suboption(DEBUG "Build with debug output" BOOL OFF)
+if(DEBUG)
+	add_definitions(-DDEBUG=1)
+endif()
+
+set(default_USE_STATIC_LIBS OFF)
+if(WIN32)
+	set(default_USE_STATIC_LIBS ON)
+endif()
+option(USE_STATIC_LIBS       "Statically link libraries" ${default_USE_STATIC_LIBS})
+option(LZMA_USE_STATIC_LIBS  "Statically link liblzma"   ${USE_STATIC_LIBS})
+option(ZLIB_USE_STATIC_LIBS  "Statically link libz"      ${USE_STATIC_LIBS})
+option(BZip2_USE_STATIC_LIBS "Statically link libbz2"    ${USE_STATIC_LIBS})
+option(Boost_USE_STATIC_LIBS "Statically link Boost"     ${USE_STATIC_LIBS})
+option(iconv_USE_STATIC_LIBS "Statically link libiconv"  ${USE_STATIC_LIBS})
+
+# Make optional dependencies required
+suboption(STRICT_USE "Abort if there are missing optional dependencies" BOOL ${CONTINUOUS_INTEGRATION})
+if(STRICT_USE)
+	set(OPTIONAL_DEPENDENCY REQUIRED)
+else()
+	set(OPTIONAL_DEPENDENCY)
+endif()
+
+# Install destinations
+if(CMAKE_VERSION VERSION_LESS 2.8.5)
+	set(CMAKE_INSTALL_DATAROOTDIR "share" CACHE
+	    STRING "read-only architecture-independent data root (share) (relative to prefix).")
+	set(CMAKE_INSTALL_BINDIR "bin" CACHE
+	    STRING "user executables (bin) (relative to prefix).")
+	set(CMAKE_INSTALL_MANDIR "${CMAKE_INSTALL_DATAROOTDIR}/man" CACHE
+	    STRING "man documentation (DATAROOTDIR/man) (relative to prefix).")
+	mark_as_advanced(
+		CMAKE_INSTALL_DATAROOTDIR
+		CMAKE_INSTALL_BINDIR
+		CMAKE_INSTALL_MANDIR
+	)
+else()
+	include(GNUInstallDirs)
+endif()
+
+
+# Helper scrips
+
+include(CheckSymbolExists)
+
+set(CMAKE_MODULE_PATH "${PROJECT_SOURCE_DIR}/cmake") # For custom cmake modules
+include(BuildType)
```

Comment:
This include can make quite a lot of changes to build config.
That clang linkage error might come from here.

Reply:
Nice, thanks! Locally it's build now. Let's look what the CI do.